### PR TITLE
Test: unit tests for IoTCertificates and IoTDeviceList visual components

### PR DESCRIPTION
**Pull request by rkpattnaik780**
### Type of change

- Unit tests
- Enhancement

### Description

This PR writes the unit tests for visual components of IoTCertificates and IoTDevice list along with taking care of null and undefined values. Some of the components and interfaces were renamed owing to #4636 .

### Checklist

- [ ] Update/write design documentation in `./documentation/design`
- [x] Write tests and make sure they pass
- [x] Resolve all warnings
- [x] Update license
- [x] Write test cases for null checks in Unit Tests
- [ ] Reference relevant issue(s) and close them after merging

### CODE REVIEW

File: console/console-init/ui/src/modules/iot-certificates/components/CertificateCard/CertificateCard.test.tsx [hunk 1]
```diff
@@ -0,0 +1,88 @@
...
+    const props: ICertificateCardProps = {
+      certificate,
+      id,
+      setOnEditMode
+    };
+
+    const { getByText } = render(<CertificateCard {...props} />);
+
+    getByText(getLabelByKey("subject-dn"));
+    certificate["subject-dn"] && getByText(certificate["subject-dn"]);
```
Comment (anukritijha) [inline console/console-init/ui/src/modules/iot-certificates/components/CertificateCard/CertificateCard.test.tsx:39]: Just curious, why is this check required?

File: console/console-init/ui/src/modules/iot-certificates/components/CertificateCard/CertificateCard.test.tsx [hunk 2]
```diff
@@ -0,0 +1,88 @@
...
+      setOnEditMode
+    };
+
+    const { getByText } = render(<CertificateCard {...props} />);
+
+    getByText(getLabelByKey("auto-provisioning-enabled"));
+    getByText("Enabled");
+  });
+
+  it("should render a card for certificate with null values", () => {
```
Comment (anukritijha) [inline console/console-init/ui/src/modules/iot-certificates/components/CertificateCard/CertificateCard.test.tsx:70]: Not sure what the difference is between this test case and the previous test case?

File: console/console-init/ui/src/modules/iot-certificates/components/CertificateCard/CertificateCard.test.tsx (hunk 1 above)
Comment (rkpattnaik780) [reply console/console-init/ui/src/modules/iot-certificates/components/CertificateCard/CertificateCard.test.tsx:39]: The argument passed to getByText should be a string only. The ensures that getByText is not called if the argument is null or undefined.

File: console/console-init/ui/src/modules/iot-certificates/components/CertificateCard/CertificateCard.test.tsx (hunk 2 above)
Comment (rkpattnaik780) [reply console/console-init/ui/src/modules/iot-certificates/components/CertificateCard/CertificateCard.test.tsx:70]: My bad, I forgot to pass null values here.

**Pull request merged**